autotest/utilities/test_ogr2ogr.py

```diff
@@ -2879,3 +2880,196 @@ def test_ogr2ogr_check_identity_transformation(ogr2ogr_path, x, y, srid):
+    field_def.SetWidth(width)
+    field_def.SetPrecision(precision)
+    lyr.CreateField(field_def)
+
+    lyr_def = lyr.GetLayerDefn()
+    field_idx = lyr_def.GetFieldIndex("num_5_2")
+    f = ogr.Feature(lyr_def)
+    f.SetField(field_idx, 12.34)
+    f.SetGeometry(ogr.CreateGeometryFromWkt("POINT(0 1)"))
+    lyr.CreateFeature(f)
+
+    f = ogr.Feature(lyr_def)
+    f.SetField(field_idx, -1.23)
+    f.SetGeometry(ogr.CreateGeometryFromWkt("POINT(0 1)"))
+    lyr.CreateFeature(f)
+
+    # This does not fit if minus sign is included in width
+    f = ogr.Feature(lyr_def)
+    f.SetField(field_idx, -12.34)
+    f.SetGeometry(ogr.CreateGeometryFromWkt("POINT(0 1)"))
+    lyr.CreateFeature(f)
+    f = None
+    lyr = None
+    ds = None
+
+    # Verify source
+    ds = ogr.Open(src_file_name)
+    lyr = ds.GetLayer(0)
+    assert lyr is not None
+    lyr_def = lyr.GetLayerDefn()
+    field_idx = lyr_def.GetFieldIndex("num_5_2")
+    field_def = lyr_def.GetFieldDefn(field_idx)
+    assert field_def.GetWidth() == width
+    assert field_def.GetPrecision() == precision
+
+    features = [f for f in lyr]
+    f = features[0]
+    assert f.GetFieldAsDouble(field_idx) == 12.34
+    f = features[1]
+    assert f.GetFieldAsDouble(field_idx) == -1.23
+    f = features[2]
+
+    if src_driver in PRECISION_UNRELIABLE_DRIVERS:
+        assert f.GetFieldAsDouble(field_idx) == -12.34
+    else:
+        assert f.GetFieldAsDouble(field_idx) == -12.3
+
+    # For CSV we want csvt to support width and precision
+    lco_options = ""
+    if dest_driver == "CSV":
+        lco_options = "-lco CREATE_CSVT=YES"
+
+    # Source is ok, convert with default options
+    gdaltest.runexternal(
+        ogr2ogr_path + f" {dest_file_name} {src_file_name} {lco_options}"
+    )
+
+    # Verify destination
+
+    # PGDump has no read capabilities, test for SQL
+    if dest_driver == "PGDump":
+        sql = ""
+        with open(dest_file_name, "r") as f:
+            sql = f.read()
+        assert "NUMERIC(4,2)" in sql
+    else:
+        dr = ogr.GetDriverByName(dest_driver)
+
+        ds = ogr.Open(dest_file_name)
+        lyr = ds.GetLayer(0)
+        assert lyr is not None
+        lyr_def = lyr.GetLayerDefn()
+        field_idx = lyr_def.GetFieldIndex("num_5_2")
+        field_def = lyr_def.GetFieldDefn(field_idx)
+        assert field_def.GetWidth() == expected_dest_width
+        assert field_def.GetPrecision() == expected_dest_precision
+
+        features = [f for f in lyr]
+        f = features[0]
+        assert f.GetFieldAsDouble(field_idx) == 12.34
+        f = features[1]
+        assert f.GetFieldAsDouble(field_idx) == -1.23
+        f = features[2]
+        if expected_dest_width == 6 or (
+            src_driver in PRECISION_UNRELIABLE_DRIVERS
+            and dest_driver in PRECISION_UNRELIABLE_DRIVERS
+        ):
+            assert f.GetFieldAsDouble(field_idx) == -12.34
+        else:
+            assert f.GetFieldAsDouble(field_idx) == -12.3
```

Comment:
to make sure the file is properly closed
```suggestion
        else:
            assert f.GetFieldAsDouble(field_idx) == -12.3
        ds = None
```